### PR TITLE
Feat: add hyperparameter definitions

### PR DESCRIPTION
**Pull request by lvyanru8200**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -42,4 +42,6 @@ go.work
 
 hack/
```
Comment (LronDC) [inline .gitignore]: hack/  是不是应该提交进去

Comment (LronDC) [reply .gitignore]: 因为后续在代码基础上 create api 时报错了
企业微信截图_f4cd6e65-839c-4390-9d7c-561fdf7dc6b8

File: api/finetune/v1beta1/finetunejob_types.go [hunk 2]
```diff
@@ -20,22 +20,81 @@ import (
...
+	Resource Resource `json:"resource"`
 }
 
 // FinetuneJobStatus defines the observed state of FinetuneJob
 type FinetuneJobStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	State FinetuneJobState `json:"state"`
+	// todo FinetuneState
+	Stats  string            `json:"stats"`
```
Comment (LronDC) [inline api/finetune/v1beta1/finetunejob_types.go:98]: what's the difference between state and stats

File: api/finetune/v1beta1/finetuneexperiment_types.go [hunk 3]
```diff
@@ -20,22 +20,26 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+type FinetuneExperimentState string
```
Comment (LronDC) [inline api/finetune/v1beta1/finetuneexperiment_types.go:24]: 没找到 FinetuneExperimentState 的值

File: .gitignore (hunk 1 above)
Comment (lvyanru8200) [reply .gitignore]: You're right, let me amend this pr

File: api/finetune/v1beta1/finetunejob_types.go (hunk 2 above)
Comment (lvyanru8200) [reply api/finetune/v1beta1/finetunejob_types.go:98]: Here's what I think:
What the status expresses is that the cr is currently in some state, e.g. running, failed
What the stats express should be:
1. ray stats, which is stored on /tmp/ray/session_xxx/usage_stats.json
2. finetuneJob processing start time and end time

File: api/finetune/v1beta1/finetuneexperiment_types.go (hunk 3 above)
Comment (lvyanru8200) [reply api/finetune/v1beta1/finetuneexperiment_types.go:24]: Oh, sorry, it's a mistake. I'll fix him. :wink:

**Comment by LronDC** [comment]
pr ready to merge, please squash commits

**Pull request merged**